src/Microsoft.Dnx.DesignTimeHost/Program.cs

```diff
@@ -119,7 +119,28 @@ public static void Main(string[] args)
 
         private static Task<Socket> AcceptAsync(Socket socket)
         {
-            return Task.Factory.FromAsync((cb, state) => socket.BeginAccept(cb, state), ar => socket.EndAccept(ar), null);
+            var tcs = new TaskCompletionSource<Socket>();
+
+            var args = new SocketAsyncEventArgs
+            {
+                AcceptSocket = socket
+            };
+
+            args.Completed += (sender, e) =>
+            {
+                if (e.SocketError == SocketError.Success)
+                {
+                    tcs.SetResult(e.ConnectSocket);
+                }
+                else
+                {
+                    tcs.SetException(new Exception($"Failed to accept connection: {e.SocketError}."));
```

Comment:
Should we use something like `InvalidOperationException` instead of the generic `Exception`?